intelmq/bots/outputs/cif3/output.py

```diff
@@ -0,0 +1,237 @@
+    """
+    Submits indicators to a CIFv3 instance
+
+
+    IntelMQ-Bot-Name: CIFv3 API
+    """
+    add_feed_provider_as_tag: bool = False
+    cif3_feed_confidence: float = 5
+    cif3_static_confidence: bool = False
+    cif3_additional_tags = []
+    cif3_token: str = None
+    cif3_url: Optional[str] = None
+    fireball: int = 500
+    http_verify_cert: bool = True
+
+    _is_multithreadable = False
+
+    def init(self):
+        cifsdk_version = int(get_cifsdk_version().get('version').split('.')[0])
+        # installed cifsdk version must be >=3 and < 4
+        if not 3<= cifsdk_version < 4:
+            HttpClient = None
+        if HttpClient is None:
+            raise MissingDependencyError(
+                'cifsdk',
+                version='3.0.0rc4,<4.0'
+            )
+        elif JsonLib is None:
+            raise MissingDependencyError(
+                'ujson',
+                version='>=2.0'
+            )
+
+        self.cif3_url = self.cif3_url.rstrip('/')
+
+        self.logger.info(f"Connecting to CIFv3 instance at {self.cif3_url!r}.")
+        self.cli = HttpClient(self.cif3_url,
+            self.cif3_token,
+            verify_ssl=self.http_verify_cert)
+
+        try:
+            _ = self.cli.ping(write=True)
+        except Exception as err:
+            raise IntelMQException(f"Error connecting to CIFv3 instance: {err}")
+        else:
+            self.logger.info('Connected to CIFv3 instance.')
+
+        self.indicator_list = []
+        self.indicator_list_max_records = self.fireball
+        self.indicator_list_max_seconds = 5
+        self.last_flushed = None
+
+    def process(self):
+        intelmq_event = self.receive_message().to_dict(jsondict_as_string=True)
+
+        cif3_indicator = self._parse_event_to_cif3(intelmq_event)
+
+        if not self.fireball:
+            self._submit_cif3_indicator(cif3_indicator)
+        elif len(self.indicator_list) > 0 and (
+            (
+                (datetime.now() - self.last_flushed).total_seconds() >
+                    self.indicator_list_max_seconds
+            ) or
+                len(self.indicator_list) >= self.indicator_list_max_records
+        ):
+            self._submit_cif3_indicator(self.indicator_list)
+            self.indicator_list.clear()
+        else:
+            if len(self.indicator_list) == 0:
+                self.last_flushed = datetime.now()
+            self.indicator_list.append(cif3_indicator)
+
+        self.acknowledge_message()
+
+    def _parse_event_to_cif3(self, intelmq_event):
+        """
+        Takes in an IntelMQ event, parses fields to those used by CIFv3
+        Returns CIFv3 Indicator object
+        """
+        # build new cif3 indicator dict
+        new_cif3_dict = {}
+        new_cif3_dict['tags'] = []
+
+        if 'feed.provider' in intelmq_event:
+            new_cif3_dict['provider'] = intelmq_event['feed.provider']
+        else:
+            new_cif3_dict['provider'] = 'IntelMQ'
```

Comment:
```suggestion
       new_cif3_dict['provider'] = intelmq_event.get('feed.provider', 'IntelMQ')
```
`intelmq_event` is just a dict, so we can use dict-like methods